packages/scripts/create-version.sh

```diff
@@ -1,86 +1,137 @@
-    git checkout -b ${NEXT_VERSION_BRANCH}
-    git add package.json package-lock.json
-    git commit -m "Bump version to ${NEXT_VERSION}"
-    git tag ${NEXT_VERSION}
+if [ -n "$PACKAGE_NAME" ]; then
+    IS_WORKSPACE_PACKAGE=1
+    PACKAGE_DIRECTORY="packages/$PACKAGE_NAME"
+    cd $PACKAGE_DIRECTORY || {
+        error "Invalid package name provided. Directory '$PACKAGE_DIRECTORY' does not exist."
+        false
+    }
+fi
+NPM_PACKAGE_NAME=`cut -d "=" -f 2 <<< $(npm run env | grep "npm_package_name")`
 
-    log "This is the last step before CI-based npm publish.";
-    log "Are you sure you want to trigger ${NPM_PACKAGE_NAME} ${NEXT_VERSION} version publish?";
-    select choice in "Yes" "No"; do
-    case "$choice" in
-        Yes)
-            log "Pushing changes...";
-            git push origin ${NEXT_VERSION_BRANCH} tag ${NEXT_VERSION}
-            git checkout ${MAIN_BRANCH}
-            break;;
-        *)
-            log "Reverting changes...";
-            git checkout ${MAIN_BRANCH}
-            git tag -d ${NEXT_VERSION}
-            git branch -D ${NEXT_VERSION_BRANCH}
-            break;;
-    esac
-    done
+MAIN_BRANCH=${MAIN_BRANCH:-master}
+log "Switching to main branch (${MAIN_BRANCH}) and checking if its up-to-date...";
+git checkout ${MAIN_BRANCH} || {
+  error "Error during checking out ${MAIN_BRANCH} branch"
+  false
+}
+git fetch || {
+  error "Error during fetching data from repository"
+  false
 }
+git pull || {
+  error "Error during pulling ${MAIN_BRANCH} branch"
+  false
+}
+git diff origin/${MAIN_BRANCH} --exit-code --compact-summary || {
+  error "Working directory not clean. Delete or push your changes to ${MAIN_BRANCH}}."
+  false
+}
+
+log "Running tests and build..."
+npm run lint --if-present || {
+  error "Running lint check failed"
+  false
+}
+npm run check-types --if-present || {
+  error "Running types check failed"
+  false
+}
+npm run test --if-present || {
+  error "Running tests failed"
+  false
+}
+npm run build --if-present || {
+  error "Running build failed"
+  false
+}
+
+NPM_VERSION_OPTIONS="--no-git-tag-version $VERSION_TYPE"
+if [[ $VERSION_TYPE == pre* ]]; then
+    NPM_VERSION_OPTIONS="$NPM_VERSION_OPTIONS --preid pre"
+fi
+echo NPM_VERSION_OPTIONS = $NPM_VERSION_OPTIONS
+npm version $NPM_VERSION_OPTIONS
+NEXT_VERSION=v$(node -p "require('./package.json').version")
+log "Creating ${NPM_PACKAGE_NAME} ${NEXT_VERSION}...";
+
+log "Creating version publish branch...";
+if [ -n "$IS_WORKSPACE_PACKAGE" ]; then
+    NEXT_VERSION_BRANCH="publish_${PACKAGE_NAME}_${NEXT_VERSION}"
+    COMMIT_MESSAGE="Bump ${PACKAGE_NAME} version to ${NEXT_VERSION}"
+    GIT_TAG="${PACKAGE_NAME}_${NEXT_VERSION}"
+else
+    NEXT_VERSION_BRANCH="publish_${NEXT_VERSION}"
+    COMMIT_MESSAGE="Bump version to ${NEXT_VERSION}"
+    GIT_TAG="${NEXT_VERSION}"
+fi
+
+git checkout -b ${NEXT_VERSION_BRANCH}
+git add package.json
+if [ -n "$IS_WORKSPACE_PACKAGE" ]; then
+    git add ../../package-lock.json
```

Comment:
That's needed as root package `package-lock.json` file changes after changing version of sub-package.